pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -56,25 +56,31 @@ type ReplicatedUpgradeReconciler struct {
 func MakeReplicatedUpgradeReconciler(vdbrecon *VerticaDBReconciler, log logr.Logger,
 	vdb *vapi.VerticaDB, pfacts *PodFacts, dispatcher vadmin.Dispatcher) controllers.ReconcileActor {
 	return &ReplicatedUpgradeReconciler{
-		VRec:       vdbrecon,
-		Log:        log.WithName("ReplicatedUpgradeReconciler"),
-		VDB:        vdb,
-		PFacts:     pfacts,
-		Manager:    *MakeUpgradeManager(vdbrecon, log, vdb, vapi.ReplicatedUpgradeInProgress, replicatedUpgradeAllowed),
+		VRec:   vdbrecon,
+		Log:    log.WithName("ReplicatedUpgradeReconciler"),
+		VDB:    vdb,
+		PFacts: pfacts,
+		Manager: *MakeUpgradeManager(vdbrecon, log, vdb, vapi.ReplicatedUpgradeInProgress,
+			replicatedUpgradeAllowed),
 		Dispatcher: dispatcher,
 	}
 }
 
 // Reconcile will automate the process of a replicated upgrade.
 func (r *ReplicatedUpgradeReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
-	if ok, err := r.Manager.IsUpgradeNeeded(ctx); !ok || err != nil {
+	sandbox := r.PFacts.getSandboxName()
+	if ok, err := r.Manager.IsUpgradeNeeded(ctx, sandbox); !ok || err != nil {
```

Comment:
When you merge up `r.PFacts` is a map now (sandbox name to podfacts). This only needs to look at the main cluster, so you probably can just pass in `vapi.MainCluster` here.
```suggestion
	if ok, err := r.Manager.IsUpgradeNeeded(ctx, vapi.MainCluster); !ok || err != nil {
```